app/models/user.rb

```diff
@@ -23,27 +23,25 @@ class User < ApplicationRecord
   # INSTANCE METHODS
   ##############################################################################################################
   def friends
-    friends_array = friendships.map { |friendship| friendship.friend_id if friendship.confirmed }
-    friends_array += inverse_friendships.map { |friendship| friendship.user_id if friendship.confirmed }
-    friends_array.compact
+    friendships.map(&:user_id) & inverse_friendships.map(&:user_id)
   end
 
   # Requests made TO current_user
   def pending_acceptance
-    inverse_friendships.map { |friendship| friendship.user_id unless friendship.confirmed }.compact
+    inverse_friendships.map(&:user_id) - friendships.map(&:user_id)
   end
```

Comment:
I notice that you choose to not implement the 'confirmed' attribute into the friendship table. This is suggested in the first Milestone:
![Screenshot from 2019-12-06 10-16-28](https://user-images.githubusercontent.com/30977845/70325757-80a48e00-1811-11ea-8eb0-1c2441fb6766.png)

Implementing the status/confirmed attribute would make easy to query requests. You can do that only with Active record, without using any ruby method.

- To get pending requests:

```ruby
has_many :pending_friendships, -> { where confirmed: false }, class_name: "Friendship", foreign_key: "user_id"
has_many :pending_friends, through: :pending_friendships, source: :friend
```
- To get confirmed friendships:
```ruby
has_many :confirmed_friendships, -> { where confirmed: true }, class_name: "Friendship"
has_many :friends, through: :confirmed_friendships
```

Please take a time to read this [article](https://explainextended.com/2009/03/07/selecting-friends/) and this StackOverflow [question](https://stackoverflow.com/questions/4219979/sql-best-practice-for-a-friendship-table)

Reply:
Fixed, we still choose not to use the confirmed flag since it's a tip from milestone 5, and milestone 6 tip is to not use it.